apps/opik-backend/src/main/java/com/comet/opik/api/Span.java

```diff
@@ -70,4 +72,11 @@ public static class Write {
         public static class Public {
         }
     }
+
+    @JsonProperty
+    @JsonView({Span.View.Public.class})
+    @Schema(accessMode = Schema.AccessMode.READ_ONLY, description = "Duration in milliseconds as a decimal number to support sub-millisecond precision")
+    public Double duration() {
+        return DurationUtils.getDurationInMillisWithSubMilliPrecision(startTime, endTime);
```

Comment:
Why aren't you calculating in ClickHouse? You can do it on the flight or add a calculated field.

Reply:
Yeah, we can. I just thought it would be simple to calculate here since we can use the values. We can add a materialized column if you think it would be a better solution

Reply:
@andrescrz done in https://github.com/comet-ml/opik/pull/889

Reply:
The important thing for the filters is to work, is that it's done in ClickHouse. It can be on the flight or as materialised column.

Reply:
The filter was tested as well, but please feel free to look over the other PR. I implement the materialised column

Reply:
Done for spans in https://github.com/comet-ml/opik/pull/883/commits/af017e0022f558f0a9ef2c6b9afbf9ab301eadca. Will do something similar for traces.